modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsSettleTimeCalculator.scala

```diff
@@ -0,0 +1,67 @@
+package seqexec.server.tcs
+
+import cats.Order
+import cats.data.NonEmptySet
+import cats.implicits._
+import mouse.boolean._
+import seqexec.model.enum.Instrument
+import seqexec.server.tcs.TcsController.{InstrumentOffset, Subsystem}
+import squants.{Ratio, Time}
+import squants.space.Angle
+import squants.time.TimeConversions._
+import squants.space.AngleConversions._
+
+object TcsSettleTimeCalculator {
+
+  trait SettleTimeCalculator {
+    def calc(displacement: Angle): Time
+  }
+
+  def constantSettleTime(cnst: Time): SettleTimeCalculator = (_: Angle) => cnst
+
+  // Settle time proportional to displacement
+  def linearSettleTime(scale: SettleTimeScale): SettleTimeCalculator = (displacement: Angle) => scale * displacement
```

Comment:
I may need to implement more formulae, like quadratic or exponential, depending on the results of measuring the stabilization times.